cgp/hl_api.py

```diff
@@ -73,15 +74,18 @@ def evolve(
         )
         termination_fitness = min_fitness
 
-    if np.isinf(max_generations) and np.isinf(max_objective_calls):
-        raise ValueError("Either max_generations or max_objective_calls must be finite.")
+    if max_generations == np.iinfo(np.int64).max and max_objective_calls == np.iinfo(np.int64).max:
+        raise ValueError(
+            "Either max_generations or max_objective_calls must be set to a"
+            " value smaller than the largest representable integer."
+        )
 
     ea.initialize_fitness_parents(pop, objective)
     if callback is not None:
         callback(pop)
 
     # perform evolution
-    max_fitness = np.finfo(np.float).min
+    max_fitness = np.finfo(np.float_).min
```

Comment:
Probably just a minor detail, but the DepreceationWarning suggested to use `np.float64` or `float` here, any specific reason why `np.float_`

Reply:
Ah I see, I didn't see that DeprecationWarning. Okay, I will use `float` instead. In fact, `np.float_` is just an alias for `np.float64`. Thanks.